src/features/ExternalApi.ts

```diff
@@ -0,0 +1,144 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+import * as vscode from "vscode";
+import { v4 as uuidv4 } from 'uuid';
+import { LanguageClient } from "vscode-languageclient";
+import { IFeature } from "../feature";
+import { Logger } from "../logging";
+import { SessionManager } from "../session";
+
+interface IExternalExtension {
+    readonly id: string;
+    readonly apiVersion: string;
+}
+
+export interface IExternalPowerShellDetails {
+    exePath: string;
+    version: string;
+    displayName: string;
+    architecture: string;
+}
+
+export class ExternalApiFeature implements IFeature {
+    private commands: vscode.Disposable[];
+    private languageClient: LanguageClient;
+    private static readonly registeredExternalExtension: Map<string, IExternalExtension> = new Map<string, IExternalExtension>();
+
+    constructor(private sessionManager: SessionManager, private log: Logger) {
+        this.commands = [
+            /*
+            DESCRIPTION:
+                Registers your extension to allow usage of the external API. The returns
+                a session UUID that will need to be passed in to subsequent API calls.
+
+            USAGE:
+                vscode.commands.executeCommand(
+                    "PowerShell.RegisterExternalExtension",
+                    "ms-vscode.PesterTestExplorer" // the name of the extension using us
+                    "v1"); // API Version.
+
+            RETURNS:
+                string session uuid
+            */
+            vscode.commands.registerCommand("PowerShell.RegisterExternalExtension", (id: string, apiVersion: string = 'v1'): string => {
+                log.writeDiagnostic(`Registering extension '${id}' for use with API version '${apiVersion}'.`);
+
+                for (const [_, externalExtension] of ExternalApiFeature.registeredExternalExtension) {
+                    if (externalExtension.id === id) {
+                        const message = `The extension '${id}' is already registered.`;
+                        log.writeWarning(message);
```

Comment:
Is there a `log.writeError` -- should we use that if we're `throw`ing?

Reply:
My thinking was that the logs are really for us, not the extension author. To _help_ the extension author a bit, I used warning... but technically it's user error, not our issue. So I felt it didn't deserve `writeError`. But happy to change it. I don't care too much.